presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -594,6 +586,11 @@ private void validateWritePageChecksum()
             if (includedColumns.containsKey(columnId)) {
                 StreamDescriptor streamDescriptor = streamDescriptors.get(columnId);
                 streamReaders[columnId] = StreamReaders.createStreamReader(streamDescriptor, hiveStorageTimeZone, systemMemoryContext);
+
+                ColumnHandle columnHandle = includedColumnHandles.get(columnId);
+                if (columnHandle != null && columnHandle.getReferencedSubfields() != null) {
+                    streamReaders[columnId].setReferencedSubfields(columnHandle.getReferencedSubfields(), 0);
```

Comment:
Should we add a default method to `setReferencedSubfields` which has depth = 0, and delegates to this version?

Reply:
Probably even better would be to push this initial variant directly into the factory method where we create the stream reader.

Reply:
@tdcmeehan Tim, I agree that we should provide all the info at the construction time and not use mutators like this. Will do that in some future refactoring.